helm-chart/binderhub/templates/ingress.yaml

```diff
@@ -5,30 +5,39 @@ metadata:
   name: binderhub
   {{- if or (and .Values.ingress.https.enabled (eq .Values.ingress.https.type "kube-lego")) .Values.ingress.annotations }}
   annotations:
-    {{- if eq .Values.ingress.https.type  "kube-lego" }}
+    {{- if and .Values.ingress.https.enabled (eq .Values.ingress.https.type "kube-lego") }}
     kubernetes.io/tls-acme: "true"
-    {{ end -}}
+    {{- end }}
     {{- range $key, $value := .Values.ingress.annotations }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
   {{- end }}
 spec:
   rules:
-    {{- range $host := .Values.ingress.hosts }}
-    - host: {{ $host }}
+    {{- range .Values.ingress.hosts }}
+    - host: {{ . }}
       http:
         paths:
           - path: /
             backend:
               serviceName: binder
               servicePort: 8585
     {{- end }}
-{{- if and .Values.ingress.https.enabled (eq .Values.ingress.https.type "kube-lego") }}
+  {{- if and .Values.ingress.https.enabled (eq .Values.ingress.https.type "kube-lego") }}
   tls:
     - secretName: kubelego-tls-binder-{{ .Release.Name }}
       hosts:
-        {{- range $host := .Values.ingress.hosts }}
-        - {{ $host }}
+        {{- range .Values.ingress.hosts }}
+        - {{ . | quote }}
+        {{- end }}
+  {{- else if .Values.ingress.tls }}
+  tls:
+    {{- range .Values.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
         {{- end }}
-{{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
```

Comment:
I added the `else if` clause to support (while being backwards compatible) the typical configuration of ingress used so often by cert-manager users where `ingress.tls` is a List containing `hosts` and `secretName`.